### PR TITLE
Upgrade ASM in Anthos Sample Deployment to 1.8.1-asm.5

### PR DESCRIPTION
**Pull request by katze120**
Upgrade ASM in Anthos Sample Deployment to 1.8.1-asm.5
* Sync-ed `asm/istio/istio-operator.yaml` and `asm/canonical-service/controller.yaml, while keeping the custom CPU requests.
* Sync-ed `Kptfile`

### CODE REVIEW

**Comment by katze120** [comment]
Question for reviewer:  Should ASD be using [identity-provider](https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/tree/release-1.8-asm/asm/identity-provider)?

File: asm/istio/istio-operator.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ kind: IstioOperator
 spec:
   profile: asm-gcp
   hub: gcr.io/gke-release/asm # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.hub"}
-  tag: 1.7.3-asm.6 # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.tag"}
+  tag: 1.8.1-asm.5 # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.tag"}
```
Comment (elfinhe) [inline asm/istio/istio-operator.yaml:20]: we'd better use `kpt set asm anthos.servicemesh.tag 1.8.1-asm.5` to also update Kptfile:115.

Comment (katze120) [reply asm/istio/istio-operator.yaml:20]: Updated in the jump server script.

**Comment by elfinhe** [comment]
> Question for reviewer: Should ASD be using [identity-provider](https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/tree/release-1.8-asm/asm/identity-provider)?

I think the answer is no, because it will be used by ASM Feature Controller. In ASD's case, feature controller is not used.

**Comment by elfinhe** [comment]
> > Question for reviewer: Should ASD be using [identity-provider](https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/tree/release-1.8-asm/asm/identity-provider)?
> 
> I think the answer is no, because it will be used by ASM Feature Controller. In ASD's case, feature controller is not used.

@JimmyCYJ to double check.

**Pull request merged**